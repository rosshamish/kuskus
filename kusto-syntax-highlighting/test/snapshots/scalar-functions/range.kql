// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/rangefunction

T | extend r = range(1, 8, 3);

T | extend r = range(datetime(2015-01-01), datetime(2015-12-31), 1d);

range(1, 3);

print r = range(1,1000000000) | mv-expand r | count
