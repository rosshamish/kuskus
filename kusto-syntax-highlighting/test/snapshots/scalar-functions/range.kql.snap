>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/rangefunction
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>T | extend r = range(1, 8, 3);
#^^^^ source.kusto
#    ^^^^^^ source.kusto keyword.other.query.kusto
#          ^^^^^ source.kusto
#               ^^^^^ source.kusto meta.scalar.function.range.kusto support.function.kusto
#                    ^ source.kusto meta.scalar.function.range.kusto
#                     ^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                      ^^ source.kusto meta.scalar.function.range.kusto
#                        ^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                         ^^ source.kusto meta.scalar.function.range.kusto
#                           ^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                            ^ source.kusto meta.scalar.function.range.kusto
#                             ^^ source.kusto
>
>T | extend r = range(datetime(2015-01-01), datetime(2015-12-31), 1d);
#^^^^ source.kusto
#    ^^^^^^ source.kusto keyword.other.query.kusto
#          ^^^^^ source.kusto
#               ^^^^^ source.kusto meta.scalar.function.range.kusto support.function.kusto
#                    ^ source.kusto meta.scalar.function.range.kusto
#                     ^^^^^^^^ source.kusto meta.scalar.function.range.kusto storage.type.kusto
#                             ^ source.kusto meta.scalar.function.range.kusto
#                              ^^^^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                                  ^ source.kusto meta.scalar.function.range.kusto
#                                   ^^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                                     ^ source.kusto meta.scalar.function.range.kusto
#                                      ^^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                                        ^ source.kusto meta.scalar.function.range.kusto
#                                         ^^ source.kusto meta.scalar.function.range.kusto
#                                           ^^^^^^^^ source.kusto meta.scalar.function.range.kusto storage.type.kusto
#                                                   ^ source.kusto meta.scalar.function.range.kusto
#                                                    ^^^^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                                                        ^ source.kusto meta.scalar.function.range.kusto
#                                                         ^^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                                                           ^ source.kusto meta.scalar.function.range.kusto
#                                                            ^^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                                                              ^ source.kusto meta.scalar.function.range.kusto
#                                                               ^^ source.kusto meta.scalar.function.range.kusto
#                                                                 ^^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                                                                   ^ source.kusto meta.scalar.function.range.kusto
#                                                                    ^^ source.kusto
>
>range(1, 3);
#^^^^^ source.kusto meta.scalar.function.range.kusto support.function.kusto
#     ^ source.kusto meta.scalar.function.range.kusto
#      ^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#       ^^ source.kusto meta.scalar.function.range.kusto
#         ^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#          ^ source.kusto meta.scalar.function.range.kusto
#           ^^ source.kusto
>
>print r = range(1,1000000000) | mv-expand r | count
#^^^^^ source.kusto keyword.other.query.kusto
#     ^^^^^ source.kusto
#          ^^^^^ source.kusto meta.scalar.function.range.kusto support.function.kusto
#               ^ source.kusto meta.scalar.function.range.kusto
#                ^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                 ^ source.kusto meta.scalar.function.range.kusto
#                  ^^^^^^^^^^ source.kusto meta.scalar.function.range.kusto constant.numeric.kusto
#                            ^ source.kusto meta.scalar.function.range.kusto
#                             ^^^ source.kusto
#                                ^^^^^^^^^ source.kusto keyword.other.query.kusto
#                                         ^^^^^ source.kusto
#                                              ^^^^^ source.kusto keyword.other.query.kusto
>